[Mellanox] Add logic to use common sai.profile per asic for mlnx and nv bluefield

noaOrMlnx opened the pull request:
Changed config_mlnx_syncd() and config_syncd_nvidia_bluefield() functionality to use per-asic sai.profile for all Spectrum{x} SKUs.
The function will apply specific SKU sai.profile, then per-asic sai.profile and lastly - sai-common.profile (which is for all Mellanox SKUs).

Why I did it
To have the ability to add common parameters to only 1 file instead of all SKUs - but per asic.

depends on https://github.com/noaOrMlnx/sonic-buildimage/pull/67 that created asic_detect.sh.

noaOrMlnx committed:
[Mellanox] Add logic to use common sai.profile per asic for mlnx and nv bluefield

oleksandrivantsiv commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -509,7 +517,21 @@ config_syncd_nvidia_bluefield()
+
+    echo >> /tmp/sai-temp.profile
+
+    DEVICE_TYPE=$(/usr/bin/asic_detect/asic_detect.sh)
```
Can you please add chekc of the return code from asic_detect.sh. To handle case when the ASIC is not detected for some reason. E.g.:
```
DEVICE_TYPE=$(/usr/bin/asic_detect/asic_detect.sh)
if [[ $? -eq 0 ]]; then
    ASIC_PROFILE_FILE="sai-${DEVICE_TYPE}.profile"
    ASIC_PROFILE_PATH="/etc/nv-bf/${ASIC_PROFILE_FILE}"
    if [ -f "$ASIC_PROFILE_PATH" ]; then
        cat "$ASIC_PROFILE_PATH" >> /tmp/sai-temp.profile
        echo >> /tmp/sai-temp.profile
    fi
else
    echo "Warning: ASIC is not detected..."
fi
```

noaOrMlnx replied:
done

noaOrMlnx committed:
Fix CR comments: use RC from asic_detect and use right folder for nv bluefield

mramezani95 committed:
Updated SAI version to 1.16.3. (#1631)

Signed-off-by: Mahdi Ramezani <mramezani@microsoft.com>

saiarcot895 committed:
Use container registry when running tests (#1635)

When running the docker-sonic-vs tests, specify a container registry to
pull containers from.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>

veronica-arista committed:
Temp workaround for sonic-buildimage issue 23387 (#1629)

Co-authored-by: StormLiangMS <89824293+StormLiangMS@users.noreply.github.com>
Co-authored-by: Ying Xie <yxieca@users.noreply.github.com>

deran1980 committed:
[Xsight] update config_syncd_xsight() (#1607)

Co-authored-by: Anton Parkhomenko <aparkhomenko@larch-networks.com>
Co-authored-by: Prince Sunny <prince.sunny@microsoft.com>
Co-authored-by: Saikrishna Arcot <sarcot@microsoft.com>

noaOrMlnx committed:
Merge branch 'master' into common-sai-profile-mlnx-bf3

dypet committed:
VPP: Update UN_PERF to UN. (#1634)

noaOrMlnx committed:
Merge branch 'master' into common-sai-profile-mlnx-bf3

nazariig committed:
[trim]: Add Packet Trimming Drop Counters to VS lib (#1630)

* [trim]: Add Packet Trimming Drop Counters to VS lib

Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

senthil-nexthop committed:
Update SAI to include macsec POST API headers (#1628)

wen587 committed:
update docker sonic test (#1639)

What I did
Install the yang related deb to make it align with the latest libswsscommon deb file.
Why I did it
The vstest will fail if not update with requirement yang file
How I verified it
Verify with vs test

noaOrMlnx committed:
Merge branch 'master' into common-sai-profile-mlnx-bf3
